content/security.md

```diff
@@ -0,0 +1,380 @@
+
+Most of your app settings should be in JSON files that you pass in when starting your app. You can start your app with a settings file by passing the `--settings` flag:
+
+```sh
+# Pass development settings when running your app locally
+meteor --settings development.json
+
+# Pass production settings when deploying your app
+meteor deploy myapp.com --settings production.json
+```
+
+Here's what a settings file with some API keys might look like:
+
+```js
+{
+  "facebook": {
+    "clientId": "12345",
+    "secret": "1234567"
+  }
+}
+```
+
+In your app's JavaScript code, these settings can be accessed from the variable `Meteor.settings`.
+
+#### Passing settings to the client
+
+In most normal situations, API keys from your settings file will only be used by the server, and by default the data passed in through `--settings` is only available on the server. However, if you put data under a special key called `public`, it will be available on the client. You might want to do this if, for example, you need to make an API call from the client. Public settings will be available on the client under `Meteor.settings.public`.
+
+#### API keys for Meteor OAuth login
+
+For the `accounts-facebook` package to pick up these keys, you need to add them to the service configuration collection in the database. Here's how you do that:
+
+First, add the `service-configuration` package:
+
+```sh
+meteor add service-configuration
+```
+
+Then, upsert into the exported collection:
+
+```js
+ServiceConfiguration.configurations.upsert(
+  { service: "facebook" },
+  {
+    $set: {
+      clientId: Meteor.settings.facebook.clientId,
+      loginStyle: "popup",
+      secret: Meteor.settings.facebook.secret
+    }
+  }
+);
+```
+
+Now, `accounts-facebook` will be able to find that API key and Facebook login will work properly.
+
+### SSL
+
+This is a very short section, but it deserves its own place in the table of contents.
+
+**Every production Meteor app that handles user data should run with SSL.**
+
+For the uninitiated, this means all of your HTTP requests should go over HTTPS, and all websocket data should be sent over WSS.
+
+Yes, Meteor does hash your password on the client before sending it over the wire, but hashing a password in this way provides only minimal security, and someone who intercepts that sent password will be able to, with time, decode it into the actual password. That's why passwords in the database are _salted_ - a random string is appended to the password before hashing. It is not possible to do this on the client, so the passwords sent over the wire are not secure even though they are hashed. The only way to secure that transfer is by using SSL.
+
+You can ensure that any unsecured connection to your app redirects to a secure connection by adding the `force-ssl` package.
```

Comment:
Also, you could mention as an aside that weird firewalls have been known to munge ws:// traffic, and wss:// seems to avoid that problem?